### PR TITLE
Adds basic PhasedISwapPow gate to Sycamore decomposition function

### PR DESCRIPTION
**Pull request by KevinVillela**
[EDITED] This algorithm works as follows: 
  - If `exponent` is 1, decompose the circuit into `ZPowGate`s and an `ISWAP`, which takes 12 one-qubit operations and 3 applications of the Sycamore gate.
  - Othwerise, if `phase_exponent` is .25, decompose the circuit into a static set of gates with coefficient, which takes 19 one-qubit operations and 4 applications of the Sycamore gate. This is the one that was in Cirq Internal.
  - If neither of these constraints are satisfied, throw an error.

### CODE REVIEW

**Comment by KevinVillela** [comment]
@mpharrigan wasn't able to request a review from you for some reason, so just mentioning instead /shrug

**Comment by mpharrigan** [comment]
@viathor @dstrain115 was this compilation lost somehow in the juggling of code from cirq-internal and removal of openfermion-cirq gates?

@ncrubin can you review this PR

File: cirq/google/optimizers/convert_to_sycamore_gates.py [hunk 1]
```diff
@@ -201,6 +212,87 @@ def known_two_q_operations_to_sycamore_operations(
...
+    """
+
+    yield ops.Z(a)**phase_exponent,
+    yield ops.Z(b)**-phase_exponent,
+    yield decompose_iswap_into_syc(a, b),
+    yield ops.Z(a)**-phase_exponent,
+    yield ops.Z(b)**phase_exponent,
+
+
+def decompose_phased_iswap_into_syc_precomputed(theta: float, a: ops.Qid,
```
Comment (ncrubin) [inline cirq/google/optimizers/convert_to_sycamore_gates.py]: Is this function covered by a test?

Comment (KevinVillela) [reply cirq/google/optimizers/convert_to_sycamore_gates.py]: I think the givens rotation does, but I could be wrong. Added an explicit test for it.

**Comment by KevinVillela** [comment]
Hmm not sure why Travis is failing, it runs fine on my machine and was fine before the sync. /shrug

**Comment by KevinVillela** [comment]
Filed https://github.com/quantumlib/Cirq/issues/2646 for the Travis failure.

**Comment by mpharrigan** [comment]
Thanks!

**Pull request merged**